tests/components/homewizard/test_sensor.py

```diff
@@ -637,3 +643,107 @@ async def test_sensor_entity_export_disabled_when_unused(
     )
     assert entry
     assert entry.disabled
+
+
+async def test_sensors_unreachable(hass, mock_config_entry_data, mock_config_entry):
+    """Test sensor handles api unreachable."""
+
+    api = get_mock_device()
+    api.data.available_datapoints = [
+        "total_power_import_t1_kwh",
+    ]
+    api.data.total_power_import_t1_kwh = 1234.123
+
+    utcnow = dt_util.utcnow()
+
+    with patch(
+        "aiohwenergy.HomeWizardEnergy",
+        return_value=api,
+    ):
+        api.update = AsyncMock(return_value=True)
+
+        entry = mock_config_entry
+        entry.data = mock_config_entry_data
+        entry.add_to_hass(hass)
+
+        await hass.config_entries.async_setup(entry.entry_id)
+        await hass.async_block_till_done()
+
+        assert (
+            hass.states.get(
+                "sensor.product_name_aabbccddeeff_total_power_import_t1"
+            ).state
+            == "1234.123"
+        )
+
+        api.update = AsyncMock(return_value=False)
+        async_fire_time_changed(hass, utcnow + timedelta(seconds=10))
+        await hass.async_block_till_done()
+        assert (
+            hass.states.get(
+                "sensor.product_name_aabbccddeeff_total_power_import_t1"
+            ).state
+            == "unavailable"
+        )
+
+        api.update = AsyncMock(return_value=True)
+        async_fire_time_changed(hass, utcnow + timedelta(seconds=10))
```

Comment:
Does it work to use the same time point two times in a row? I don't remember if that's robust or not.

Reply:
I am a bit confused.

If I look a bit around the test that use `async_fire_time_changed`, some use a fixed `utcnow + timedelta` and some use `utcnow() + timedelta`, see [here](https://github.com/home-assistant/core/blob/0407f60955c24b33dd12841107ce055bb4bc6d05/tests/components/aurora_abb_powerone/test_sensor.py#L127-L152) and [here](https://github.com/home-assistant/core/blob/0407f60955c24b33dd12841107ce055bb4bc6d05/tests/components/asuswrt/test_sensor.py#L196-219).

What do you prefer?
```diff
utcnow = dt_util.utcnow()
async_fire_time_changed(hass, utcnow + timedelta(seconds=5))
....
async_fire_time_changed(hass, utcnow + timedelta(seconds=10))
```
or 
```diff
utcnow = dt_util.utcnow()
async_fire_time_changed(hass, utcnow() + timedelta(seconds=5))
....
async_fire_time_changed(hass, utcnow() + timedelta(seconds=5))
```

Or is it that I just don't understand what you want 🤷

Reply:
I prefer the former.

Reply:
All right. Rolled back the change.

Reply:
This is the former:
```
utcnow = dt_util.utcnow()
async_fire_time_changed(hass, utcnow + timedelta(seconds=5))
....
async_fire_time_changed(hass, utcnow + timedelta(seconds=10))
```

Reply:
👍 Hope we are on the same page now.

Reply:
Yes, now it looks good!